How deadlines and transient retries interact (#24218)
dotnet/AspNetCore.Docs: aspnetcore/grpc/deadlines-cancellation.md

[!code-csharp[](~/grpc/deadlines-cancellation/deadline-server.cs?highlight=5)]

### Deadlines and retries

> When a gRPC call is configured with [retry fault handling](xref:grpc/retries) and a deadline, the deadline tracks time across all retries for a gRPC call. When the deadline is exceeded, a gRPC call immediately aborts the underlying HTTP request, skips any remaining retries, and throws a `DeadlineExceeded` error.

Author: ```suggestion
When a gRPC call is configured with [retry fault handling](xref:grpc/retries) and a deadline, the deadline tracks time across all retries for a gRPC call. If the deadline is exceeded, a gRPC call immediately aborts the underlying HTTP request, skips any remaining retries, and throws a `DeadlineExceeded` error.
```

> gRPC supports [transient fault handing using retries](xref:grpc/retries). A gRPC call can be configured with retries and a deadline. In that situation a deadline tracks time across all retries for a gRPC call. When the deadline is exceeded a gRPC call immediately aborts the underlying HTTP request, skips any remaining retries, and throws a `DeadlineExceeded` error.

Author: ```suggestion
When a gRPC call is configured with [retry fault handling](xref:grpc/retries) and a deadline, the deadline tracks time across all retries for a gRPC call. When the deadline is exceeded a gRPC call immediately aborts the underlying HTTP request, skips any remaining retries, and throws a `DeadlineExceeded` error.
```
